pkg/recovery/recovery.go

```diff
@@ -12,34 +12,48 @@ import (
 	"github.com/appscode/stash/pkg/cli"
 	"github.com/appscode/stash/pkg/eventer"
 	"github.com/appscode/stash/pkg/util"
+	"github.com/cenkalti/backoff"
```

Comment:
Can you use the `wait` library from Kubernetes? 
https://github.com/kubernetes/apimachinery/blob/master/pkg/util/wait/wait.go#L192

Reply:
Arguably it's slightly less nice in terms of UX to have retries vs. deadline in this case. You think the "max-steps" would make a better end-user experience?

Note that `cenkalti/backoff` isn't introducing a new dependency.

Reply:
We used to use `cenkalti/backoff` library. Now, these days we try to use the `wait` util .

I think there should be a timeout (however large). Otherwise, how does user know that the job is failing?

Reply:
Do you want me to keep the `backoff max wait` (in seconds) UI and calculate the steps in code for that?

Reply:
> Otherwise, how does user know that the job is failing?

If the job didn't succeed then it failed to deliver the restore. I'd expect the end users to monitor for the restore to finish in some amount of time before getting paged. Keeping the pod "stuck forever" would actually help to find the source of the problem, given that the network stack would be pre-configured for that specific failing condition.

Reply:
The more I look into kube's backoff code the less I like it, really. There's no straightforward way to use it with `context.WithTimeout` to set up a deadline. It requires you to specify the number of retries which is abstract if you cannot specify the factor, so user will have to specify both the factor and the retries count to allow for the proper flexibility (or to set the retry count really high and break the loop via other means).

I've looked around for external references to that code and they aren't really motivating either. https://sourcegraph.com/github.com/tnozicka/openshift-kubernetes@5ada62a9f5a4f36199e94d64aa1698386935526a/-/blob/pkg/cloudprovider/providers/aws/aws.go#L142 uses fixed constants and approximations to e.g. get to 30 minutes.

I stand by my design idea of that the user should configure the deadline, just like you'd do with other sane RPC implementations.

Reply:
Sounds good to me.